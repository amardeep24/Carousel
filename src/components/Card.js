import React from 'react'
import PropTypes from 'prop-types'
import { formatCurrency } from "../utils/utils";
import {
    CardContainer,
    CardHeader,
    CardBody,
    CardFooter,
    CardImage
} from './Styled';

function Card({ title, subHeader, header, footer, active }) {
    return (
        <CardContainer active={active}>
            <CardHeader>
                <CardImage src={header} />
                {formatCurrency(subHeader)}
            </CardHeader>
            <CardBody>
                {title}
            </CardBody>
            <CardFooter>
                {footer}
            </CardFooter>
        </CardContainer>
    )
}

Card.propTypes = {
    title: PropTypes.string,
    subHeader: PropTypes.string,
    header: PropTypes.string,
    footer: PropTypes.string,
    active: PropTypes.bool
}

export default Card

