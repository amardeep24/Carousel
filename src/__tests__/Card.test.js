import React from 'react';
import { shallow, configure } from 'enzyme';
import Card from '../components/Card';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

describe('Test 1::The card component should match the snapshot', () => {
    let component;

    beforeEach(() => {
        component = shallow(<Card
            title="foo"
            subHeder={12000}
            header="bar"
            footer="dummy"
        />);
    });

    it('should match snapshot', () => {
        expect(component).toMatchSnapshot();
    });
});

describe('Test 2::The card component should render children properly', () => {
    let component;

    beforeEach(() => {
        component = shallow(<Card
            title="title"
            subHeader={12000}
            header="header"
            footer="footer"
        />);
    });

    it('should have 3 children', () => {
        expect(component.dive().find('div').children()).toHaveLength(3);
    });

    it('should render the title, subHeader, footer correctly', () => {
        expect(component.dive().find('div').childAt(0).text()).toEqual("â‚¹12,000.00");
        expect(component.dive().find('div').childAt(1).text()).toEqual("title");
        expect(component.dive().find('div').childAt(2).text()).toEqual("footer");
    });
});
