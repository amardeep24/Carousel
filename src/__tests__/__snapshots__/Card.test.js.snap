// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 1::The card component should match the snapshot should match snapshot 1`] = `ReactWrapper {}`;
